corehq/apps/sso/forms.py

```diff
@@ -618,6 +657,52 @@ def get_primary_fields(self):
             ),
         ]
 
+    def get_remote_user_management_fields(self):
+        masked_api = get_masked_string(self.idp.api_secret)
+
+        api_secret_toggles = crispy.Div(
+            crispy.HTML(
+                format_html(
+                    '<p class="form-control-text" data-bind="hidden: isAPISecretVisible">{} '
+                    '<a href="#" data-bind="click: showAPISecret">{}</a></p>',
+                    masked_api,
+                    gettext("Update Secret")
+                ),
+            ),
+            crispy.HTML(
+                format_html(
+                    '<p class="form-control-text" data-bind="visible: isCancelUpdateVisible">'
+                    '<a href="#" data-bind="click: hideAPISecret">{}</a></p>',
+                    gettext("Cancel Update")
+                ),
+            ),
+            style="display: none;",  # prevent html showing before knockout is executed, will set visible to false
+            data_bind="visible: true",
+        )
+        return [crispy.Div(
+            crispy.Div(
+                crispy.Fieldset(
+                    _('Remote User Management'),
+                    twbscrispy.PrependedText('enable_user_deactivation', ''),
+                    'api_host',
+                    'api_id',
+                    hqcrispy.B3MultiField(
+                        gettext("Client Secret"),
+                        crispy.Div(
+                            hqcrispy.InlineField(
+                                'api_secret',
+                                data_bind="visible: isAPISecretVisible"
+                            ),
+                            api_secret_toggles,
+                        ),
+                        show_row_class=False,
+                    ),
+                    crispy.Field('date_api_secret_expiration', css_class='date-picker'),
```

Comment:
regarding my comment about using knockout bindings instead of `getElementById`
```suggestion
                    crispy.Field('date_api_secret_expiration', css_class='date-picker', data_bind="text: dateApiSecretExpiration"),
```

and then in the knockout model
```
self.dateApiSecretExpiration = ko.observable();
```

Reply:
Fixed here: ccf9b5d6b6a08df2a665edfa21636d30d6128875